### ISSUE TITLE
Feature Request: Allowing Attributes to be mapped in urlSync

### ISSUE THREAD

**Issue by Ardakilic** [algolia/instantsearch.js#1799]
Hello,

I'm using Algolia on a project that I'm working with, but I've struggled on a step. I hope you guys can assist me here:

Currently, this is the decoded url structure of my search parameter:

```js
urlSync: {
		useHash: false,
		mapping: {
		  q: 'ara',
		  p: 'sayfa'
		},
		trackedParameters: [
			'query', //q
			'attribute:color',
			'attribute:brand',
			'page' //p
		]
  },

```
`ara=query&sayfa=0&dFR[brand][0]=brandname&dFR[color][0]=colorname&is_v=1`

But for a cleaner structure, I want to achieve something like this:

`ara=query&sayfa=0&brand=brandname&color=colorname&is_v=1`

And for multiple occasions, maybe something like this:

`ara=query&sayfa=0&brand=brandname1,brandname2&color=colorname1,colorname2&is_v=1`

So in short, how do I change `&dFR[attribute][0]=value` to `&attribute=value`, at least to `&attribute[0]=value`

I'm using instantsearch.js 1 from cdn in pro plan, if that would matter.

Any idea would be appreciated.

Thanks in advance,

**Comment by bobylito** [comment]
Hi @Ardakilic Sorry we don't offer such flexibility for now :( However we're thinking hard on how it would be best to add this without crippling the API.

**Comment by smtaydemir** [comment]
need this feature, +1

**Comment by oguzhanaslan** [comment]
this feature important +1

**Comment by nolawill** [comment]
+1!
Be a huge value to have flexibility of options for masking the url params some way.

**Comment by bobylito** [comment]
Now that we've released we are going to be able to work on this kind of feature.

**Comment by bobylito** [comment]
This will be solved with #2216 and more specifically an API like the proposed in #2636. Have a look, there are demos with the source code.

Closing this for now.
